### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by jimmybw**

### CODE REVIEW

File: slack-go.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+define("GOSERVER", "http://go.brandwatch.com/");
+define("GOSERVERSHORT", "http://go/");
+define("GOSERVERDISPLAY", "go");
```
Comment (bw-allardvanhelbergen) [inline slack-go.php]: Could we make this `Go/`? I would like the official title to have a capital and the slash everywhere.

File: slack-go.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+define("BOTNAME", "go");
```
Comment (bw-allardvanhelbergen) [inline slack-go.php]: Could we make this Go/? I would like the official title to have a capital and the slash everywhere.

File: slack-go.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+define("GOSERVER", "http://go.brandwatch.com/");
```
Comment (bw-allardvanhelbergen) [inline slack-go.php]: Constants should be in SCREAMING_SNAKE_CASE.
https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-1-basic-coding-standard.md#41-constants

File: slack-go.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    $options = array( 
+        CURLOPT_RETURNTRANSFER => true,     // return web page 
+        CURLOPT_HEADER         => false,    // return headers 
+    ); 
+ 
+    $ch = curl_init( GOSERVER . '_api/go-link.json' ); 
+    curl_setopt_array( $ch, $options ); 
+    $content = curl_exec( $ch ); 
+ 
+    if(curl_errno($ch)) {
```
Comment (bw-allardvanhelbergen) [inline slack-go.php]: Missing space after `if`

File: slack-go.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+define("GOSERVER", "http://go.brandwatch.com/");
+define("GOSERVERSHORT", "http://go/");
+define("GOSERVERDISPLAY", "go");
+define("BOTNAME", "go");
+define("INCOMINGWEBHOOK", "https://hooks.slack.com/services/xxx"); //Use your own incoming webhook URL here
+
+function get_long_url( $slug )
```
Comment (bw-allardvanhelbergen) [inline slack-go.php]: Function names in camelCase.
https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-1-basic-coding-standard.md#1-overview

File: slack-go.php [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+define("INCOMINGWEBHOOK", "https://hooks.slack.com/services/xxx"); //Use your own incoming webhook URL here
+
+function get_long_url( $slug ) 
+{ 
+    $options = array( 
+        CURLOPT_RETURNTRANSFER => true,     // return web page 
+        CURLOPT_HEADER         => false,    // return headers 
+    ); 
+ 
+    $ch = curl_init( GOSERVER . '_api/go-link.json' );
```
Comment (bw-allardvanhelbergen) [inline slack-go.php]: I couldn't decipher what `$ch` stands for. Would it be possible to have a more descriptive name for this variable?

Comment (bw-allardvanhelbergen) [inline slack-go.php]: `GOSERVER . '_api/go-link.json'` should probably be a constant, no?

File: slack-go.php [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    $links = json_decode($content, true);
+
+    foreach ($links as $link) {
+    	if ($link['shortUri'] == $slug) {
+            return $link['longUri'];
+    	}
+    }
+    die ("No link found for " . $slug);
+}
+
```
Comment (bw-allardvanhelbergen) [inline slack-go.php:43]: I gather that this is where the integration actually starts doing things, but I can't quite follow what and how. Would it be possible to break things up into more functions with one main() for instance? E.g. perhaps the second curl could be put into its own function.

File: slack-go.php [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+$long_url = get_long_url( $_POST["text"] );
+
+// do the incoming webhook
+$payload = array(
+    text => "@" . $_POST["user_name"] . " shared *<" . GOSERVERSHORT . $_POST["text"] . "|" . GOSERVERDISPLAY . "/" . $_POST["text"] . ">*\n" . "_ <" . GOSERVERSHORT . $_POST["text"] . "|" . urlencode($long_url) . ">_",
```
Comment (bw-allardvanhelbergen) [inline slack-go.php]: This not very readable and it's also a very long line. Perhaps break it up over multiple lines?
https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#1-overview

**Comment by bw-allardvanhelbergen** [comment]
Hey @jimmybw,

I don't remember any of the minutia of PHP any more, so I grabbed the first significant code styleguide I could find (http://www.php-fig.org/) and used that as a basis to a work from. Most comments are about stylistic issues which apply throughout, but there's a few about structure as well.

Cheers

File: slack-go.php [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    username => BOTNAME,
+    channel => $_POST["channel_id"],
+    link_names => 1,
+    mrkdwn => "true"
+);
+
+$json_payload = json_encode($payload);
+
+$ch = curl_init();
+curl_setopt($ch,CURLOPT_URL, INCOMINGWEBHOOK);
```
Comment (bw-allardvanhelbergen) [inline slack-go.php]: Missing space after `,`.

File: slack-go.php (hunk 2 above)
Comment (jimmybw) [inline slack-go.php]: Fixed

File: slack-go.php (hunk 3 above)
Comment (jimmybw) [inline slack-go.php]: Fixed

File: slack-go.php (hunk 4 above)
Comment (jimmybw) [inline slack-go.php]: Fixed

File: slack-go.php (hunk 5 above)
Comment (jimmybw) [inline slack-go.php]: Fixed

File: slack-go.php (hunk 7 above)
Comment (jimmybw) [inline slack-go.php:43]: I broke some things out into functions, maybe it's clearer?

File: slack-go.php (hunk 9 above)
Comment (jimmybw) [inline slack-go.php]: Fixed

**Comment by jimmybw** [comment]
I reworked this a bit, feel free to take a look @bw-allardvanhelbergen

File: slack-go.php (hunk 7 above)
Comment (bw-allardvanhelbergen) [inline slack-go.php:43]: Yeah, that's much clearer now. :+1:

**Comment by bw-allardvanhelbergen** [comment]
Awesome! :fireworks: 

LGTM

**Pull request merged**